command_runner/runner.go

```diff
@@ -29,14 +30,22 @@ type Runner interface {
 	RunSubCmdByName(cmdName string, subCommand string, c *cli.Context) (err error)
 }
 
+type Factory interface {
```

Comment:
Unfortunately, they are required to avoid circular dependencies which prevent it from compiling.